app/src/main/java/net/gsantner/markor/util/AppSettings.java

```diff
@@ -325,10 +325,69 @@ public void addRecentDocument(File file) {
         ShortcutUtils.setShortcuts(_context);
     }
 
+    public void copyFavouriteFile(File fileBefore, File fileAfter) {
+        List<String> list = new ArrayList<>();
+        List<File> favourites = getFavouriteFiles();
+        for (File f : favourites) {
+            // Only existing files or file which is 'virtual storage' is considered
+            if (f != null && (f.exists() || FilesystemViewerAdapter.isVirtualStorage(f))) {
+                list.add(f.getAbsolutePath());
+            }
+        }
+        String abs = fileBefore.getAbsolutePath();
+        boolean starred = false;
+        if (list.contains(abs)) {
+            starred = true;
+        }
+        if (starred) {
+            abs = fileAfter.getAbsolutePath();
+            if (!list.contains(abs)) {
+                list.add(abs);
+            }
+        }
+        setStringList(R.string.pref_key__favourite_files, list);
+    }
+
+    public void deleteFavouriteFile(File file) {
```

Comment:
the get favourites function automatically scans all favorites for existance (as files can be deleted also in other apps without Markor). So we don't need a function for this

Reply:
The reason I wrote this method is I wished only to modify status after files are successfully deleted(since deletions culd fail and in such cases I think it might be better to have the favourites un-modified). So after deletion, this method is called upon to deal with those files still recorded as favourite but is already deleted.